src/backend/session/libseat.rs

```diff
@@ -193,6 +193,24 @@ impl Session for LibSeatSession {
 }
 
 impl LibSeatSessionNotifier {
+    fn dispatch(&mut self) {
+        self.internal.seat.borrow_mut().dispatch(0).unwrap();
+
+        for signal in self.rx.try_iter() {
+            match signal {
+                SeatEvent::Enable => {
+                    self.internal.active.store(true, Ordering::SeqCst);
+                    self.signaler.signal(SessionSignal::ActivateSession);
+                }
+                SeatEvent::Disable => {
+                    self.internal.active.store(false, Ordering::SeqCst);
+                    self.signaler.signal(SessionSignal::PauseSession);
+                    self.internal.seat.borrow_mut().disable().unwrap();
+                }
+            }
+        }
+    }
+
     /// Creates a new session object belonging to this notifier.
```

Comment:
I think this should be using a `calloop` channel, and have it registered as a sub-event source to calloop, so that it can also wake up the event source if necessary, given the whole point is that sometimes libseat can invoke the callback outside of its `dispatch()` method.